pkg/gather/ssh/ssh.go

```diff
@@ -13,10 +19,20 @@ import (
 	"github.com/sirupsen/logrus"
 	"golang.org/x/crypto/ssh"
 	"golang.org/x/crypto/ssh/agent"
+	"golang.org/x/net/proxy"
 
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 )
 
+// httpProxy is a HTTP/HTTPS connect proxy.
+type httpProxy struct {
+	host     string
+	haveAuth bool
+	username string
+	password string
+	forward  proxy.Dialer
```

Comment:
can we use proxy.FromEnvironment? It seems to return a Dialer as well and might avoid the getHttpProxyUrl func

Reply:
So now we are providing the URL directly, so we can't use the Environment.
However I checked if we could use `proxy.FromURL` to return their Dialer, but this doesn't support HTTP Proxy (only Socks) - which is why we have to have the custom struct and function:
https://cs.opensource.google/go/x/net/+/refs/tags/v0.2.0:proxy/proxy.go;l=81-110

I'll close the comment but please re-open if you still think we can re-use something.